eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -121,25 +139,49 @@ jobs:
         vcpkg.deps: 'curl[ssl] paho-mqtt cmocka'
         VCPKG_DEFAULT_TRIPLET: 'x64-linux'
         build.args: ' -DTRANSPORT_CURL=ON -DTRANSPORT_PAHO=ON -DAZ_PLATFORM_IMPL=POSIX -DUNIT_TESTING=ON'
+      Linux_x64_Release_with_samples_and_unit_test:
```

Comment:
The goal is to try to reduce the CI matrix and not add too much CI/build time overhead but still get higher coverage. 

We should look at the existing matrix, and see which ones we can flip to release, without losing coverage, and avoid adding too many new ones.

In fact, we should default to release for all, and pick a few to remain debug.

Reply:
If we want to reduce CI pipelines, let's merge build_only with build_wiht_unit_tests.
After we call generate and build (for build only) we can call cmake generate  again with tests. Then run the tests.
This would use incremental build from cmake and build faster the second time.